### ISSUE TITLE
oxfmt(sort-imports): tiebreak by import kind (value vs type)

### ISSUE THREAD

**Issue by MatissJanis** [oxc-project/oxc#22856]
**Disclaimer: AI generated, but heavily reviewed/redacted by myself**. I am not sure if this better fits as a feature request or bug report. Classifying as a bug report for now as it feels like an _unexpected behaviour_.

### Summary

`experimentalSortImports` sorts imports by `(group, module path)`. When two import statements have the **identical** module specifier — typically a value import and a separate `import type` from the same module — there is no secondary sort key, so their relative order is left to the original (stable) input order. As a result, the two valid orderings below are both treated as "sorted", and neither is normalized to the other:

```ts
// both of these are left untouched by oxfmt
import { styles } from '@actual-app/components/styles';
import type { CSSProperties } from '@actual-app/components/styles';
```

```ts
import type { CSSProperties } from '@actual-app/components/styles';
import { styles } from '@actual-app/components/styles';
```

This means a contributor can flip the order of a value import and its same-module `import type` and it passes `oxfmt --check` / CI, producing avoidable diff churn and inconsistency across a codebase.

### Real-world example

This came up reviewing a PR in actualbudget/actual (https://github.com/actualbudget/actual/pull/7922/changes#diff-58675f59248b1aae13bbaa30bef704ff6756a7ad0d9f8621ab256839f80bdea2): a contributor reordered a value import and its same-module `import type`, and nothing in `oxfmt --check` or the lint hooks flagged or fixed it. We'd like the formatter to normalize this automatically.

### Why the existing options don't solve it

Splitting into kind-specific groups (`value-external` then `type-external`) *does* make the order deterministic, but it **clusters all type imports to the end of the group**, which pulls a type import away from its same-module value import. For example, this:

```ts
import { resolveConfig } from '#config';
import type { CliConfig } from '#config';
import { withConnection } from '#connection';
```

becomes:

```ts
import { resolveConfig } from '#config';
import { withConnection } from '#connection';
import type { CliConfig } from '#config';   // separated from its '#config' value import
```

We want same-module imports to stay **adjacent**, just in a consistent order — which the group mechanism can't express.

**Comment by leaysgur** [comment]
> I am not sure if this better fits as a feature request or bug report. Classifying as a bug report for now as it feels like an unexpected behaviour.

I've determined this is a feature request.

My reasoning is:
- `perfectionist/sort-imports` also behaves this way by default
  - Specifying subgroups in `groups` does not affect the sorting order
- However, `perfectionist/sort-imports` does have an option called `fallbackSort`
  - Setting `fallbackSort: { type: 'subgroup-order' }` serves exactly this purpose
  - refs: https://github.com/azat-io/eslint-plugin-perfectionist/issues/671
- While we could implement this ourselves
  - But we aren't aiming for a 100% compatibility
- We would still need to discuss whether just `unsorted` and `subgroup-order` are sufficient, or if it should behave like this by default, etc...
